app/forms/declaration/additionaldocuments/DocumentsProduced.scala

```diff
@@ -56,28 +57,43 @@ object DocumentsProduced extends DeclarationPage {
   val documentStatusReasonKey = "documentStatusReason"
   val issuingAuthorityNameKey = "issuingAuthorityName"
   val dateOfValidityKey = "dateOfValidity"
-  val documentWriteOffKey = "documentWriteOff"
 
   val mapping = Forms
     .mapping(
-      documentTypeCodeKey -> optional(text().verifying("supplementary.addDocument.documentTypeCode.error", hasSpecificLength(4) and isAlphanumeric)),
+      documentTypeCodeKey -> optional(text().verifying("declaration.addDocument.documentTypeCode.error", hasSpecificLength(4) and isAlphanumeric)),
       documentIdentifierKey -> optional(
-        text().verifying(
-          "supplementary.addDocument.documentIdentifier.error",
-          nonEmpty and isAlphanumericWithAllowedSpecialCharacters and noLongerThan(35)
-        )
+        text()
+          .verifying("declaration.addDocument.documentIdentifier.error", nonEmpty and isAlphanumericWithAllowedSpecialCharacters and noLongerThan(35))
       ),
-      documentStatusKey -> optional(text().verifying("supplementary.addDocument.documentStatus.error", noLongerThan(2) and isAllCapitalLetter)),
+      documentStatusKey -> optional(text().verifying("declaration.addDocument.documentStatus.error", noLongerThan(2) and isAllCapitalLetter)),
       documentStatusReasonKey -> optional(
-        text().verifying("supplementary.addDocument.documentStatusReason.error", noLongerThan(35) and isAlphanumeric)
+        text().verifying("declaration.addDocument.documentStatusReason.error", noLongerThan(35) and isAlphanumeric)
       ),
       issuingAuthorityNameKey -> optional(
         text()
-          .verifying("supplementary.addDocument.issuingAuthorityName.error.length", noLongerThan(issuingAuthorityNameMaxLength))
+          .verifying("declaration.addDocument.issuingAuthorityName.error.length", noLongerThan(issuingAuthorityNameMaxLength))
       ),
       dateOfValidityKey -> optional(Date.mapping),
       documentWriteOffKey -> optional(DocumentWriteOff.mapping)
     )(DocumentsProduced.apply)(DocumentsProduced.unapply)
 
   def form(): Form[DocumentsProduced] = Form(mapping)
+
+  def fieldGroupValidation(form: Form[DocumentsProduced]): Form[DocumentsProduced] = {
+
+    def validate(docs: DocumentsProduced) =
+      docs.documentWriteOff.map(wo => DocumentWriteOff.validate(wo)).getOrElse(Seq.empty)
+
+    def withMeasurementUnitGroupErrors(formErrors: Seq[FormError]) =
+      if (formErrors.exists(_.key.endsWith(measurementUnitKey)) && formErrors.exists(_.key.endsWith(qualifierKey)))
+        Seq(
+          FormError(s"$documentWriteOffKey.$measurementUnitKey", "declaration.addDocument.measurementUnitAndQualifier.error"),
+          FormError(s"$documentWriteOffKey.$qualifierKey", "")
+        ) ++ formErrors.filter(err => !err.key.endsWith(measurementUnitKey) && !err.key.endsWith(qualifierKey))
```

Comment:
`Probably it will be worth to put it in separate variables. val additionalErrors, val filteredErrors additionalErrors ++ filteredErrors. But it's not that important, can be more readable

Reply:
OK